### PR TITLE
Journalist can submit an article to the admin site

### PR DESCRIPTION
**Pull request by Fraseer**
PT: https://www.pivotaltracker.com/story/show/179904975

## User Story
```
As a journalist
In order to publish my work
I would like to submit my articles to the admin site
```

## Tasks
- Create Input Field for the following things: title, journalist, lede, category, date, body
- Create `POST` request to API
- Create API Helper file
- Happy path only

## Screenshot

image

### CODE REVIEW

File: cypress/fixtures/example.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (maxarvid) [inline cypress/fixtures/example.json]: Is this example file needed?

File: cypress/integration/userCanCreateArticle.feature.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+describe("User can create an article", () => {
+  beforeEach(() => {
+    cy.intercept("POST", "**/api/articles**", {
+      message: "Your article has been successfully created",
```
Comment (maxarvid) [inline cypress/integration/userCanCreateArticle.feature.js]: The API responds with the following message: `You have successfully added #{title} to the site`

File: cypress/integration/userCanCreateArticle.feature.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  });
+
+  it("is expected input the following data", () => {
+    cy.get("[data-cy=title-input]").type("My Title");
+    cy.get("[data-cy=journalist-input]").type("Frank FakeName");
+    cy.get("[data-cy=lede-input]").type("What going on");
+    cy.get("[data-cy=category-input]")
+      .click()
+      .within(() => {
+        cy.contains("Business").click();
```
Comment (maxarvid) [inline cypress/integration/userCanCreateArticle.feature.js:24]: nice solution!

File: cypress/integration/userCanCreateArticle.feature.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      });
+    cy.get("[data-cy=body-input]").type("Best story in the world");
+    cy.get("[data-cy=submit-button]").click();
+    cy.get("[data-cy=article-creation]").should(
+      "contain",
+      "Your article has been successfully created"
```
Comment (maxarvid) [inline cypress/integration/userCanCreateArticle.feature.js]: See above, I believe the API will have the article title included in the response

File: src/components/CreateArticle.jsx [hunk 5]
```diff
@@ -1,11 +1,111 @@
...
+    register("category");
+    register("body");
+    // eslint-disable-next-line
+  }, []);
+
+  const { register, handleSubmit, setValue } = useForm();
+  const onSubmit = (article) => {
+    Article.create({ article }).then((response) => {
+      if (
+        response.data.message === "Your article has been successfully created"
```
Comment (maxarvid) [inline src/components/CreateArticle.jsx]: Same as above to get admin frontend in line with API

File: package.json [hunk 6]
```diff
@@ -6,10 +6,15 @@
     "@testing-library/jest-dom": "^5.11.4",
     "@testing-library/react": "^11.1.0",
     "@testing-library/user-event": "^12.1.10",
+    "lodash": "^4.17.21",
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
+    "react-hook-form": "^7.17.5",
+    "react-redux": "^7.2.5",
```
Comment (maxarvid) [inline package.json]: I think you can remove redux here

File: package.json [hunk 7]
```diff
@@ -6,10 +6,15 @@
...
     "react-router-dom": "^5.3.0",
     "react-scripts": "4.0.3",
+    "react-semantic-ui-datepickers": "^2.15.0",
+    "redux": "^4.1.1",
```
Comment (maxarvid) [inline package.json]: As above

File: cypress/fixtures/example.json (hunk 1 above)
Comment (dorianbuck) [reply cypress/fixtures/example.json]: It's been removed

File: cypress/integration/userCanCreateArticle.feature.js (hunk 2 above)
Comment (dorianbuck) [reply cypress/integration/userCanCreateArticle.feature.js]: This has been adjusted

**Review by maxarvid — APPROVED** [review]
GLTM

**Review by lukeperrin10 — APPROVED** [review]

**Pull request merged**